### ISSUE TITLE
smolvla can not reproduce results in libero eval using official checkpoints

### ISSUE THREAD

**Issue by zouyuanpeng** [huggingface/lerobot#3264]
### Environment & System Info
ubuntu2204 nvidia rtx5090D, cuda12.8
lerobot 0.5.1

### Description

Hi, I eval smolvla in libero with offical ckpt: https://huggingface.co/HuggingFaceVLA/smolvla_libero ,and the results are far from that in paper.
My result is below:
libero_object	93.0%	0.93	
libero_goal	81.0%	0.81	
libero_spatial	63.0%	0.63	
libero_10	       56.0%	0.56	
overall          73.25%     0.7325
my lerobot version is 0.5.1, mujoco version is 3.3.2, and i use libero v3.0 dataset from https://huggingface.co/datasets/HuggingFaceVLA/libero, as recommend.

my eval command is:

export HF_LEROBOT_HOME="/home/zouyp12/data/pub"
export HF_HUB_OFFLINE=1
export TRANSFORMERS_OFFLINE=1
export HF_DATASETS_OFFLINE=1

lerobot-eval --policy.path=/home/zouyp12/projects/vlas/pretrained_models/smolvla_libero \
            --policy.num_steps=10 \
            --policy.n_action_steps=10 \
            --env.type=libero \
            --env.task=libero_spatial,libero_object,libero_goal,libero_10 \
            --eval.n_episodes=10 \
            --eval.batch_size=1 \
            --policy.device=cuda \
            --output_dir=/home/zouyp12/projects/vlas/lerobot_workspace/smolvla_libero_eval

Could someone take a look and see if there are any issues?

**Comment by rylinjames** [comment]
your numbers look low but the eval config might be the issue. a few things to check:

`n_episodes=10` is very low for Libero -- the paper likely used 50+ episodes per task for stable numbers. at 10 episodes the variance alone can swing 15-20% on libero_spatial and libero_10. try bumping to 50 before concluding the checkpoint is underperforming.

also check `n_action_steps`. the default in lerobot is often mismatched from what the model was trained with. SmolVLA was trained with a specific chunk size and if your eval is using a different value the success rate drops significantly.

libero_10 at 56% specifically suggests the model is struggling with longer horizon tasks which is consistent with an action chunk size mismatch rather than a model quality issue.

**Comment by jashshah999** [comment]
Reproducibility issues can sometimes come from data quality differences. [lerobot-doctor](https://github.com/jashshah999/lerobot-doctor) can verify that your local dataset matches expected properties:

```bash
lerobot-doctor /path/to/libero_dataset --checks metadata,statistics,consistency
```

Checks for: distribution shift, feature schema mismatches, stats.json drift from actual data, and episode health issues.

https://github.com/jashshah999/lerobot-doctor

**Comment by rylinjames** [comment]
the libero_10 drop specifically might be a known eval bug — lerobot #2850. the batch_size parameter caps how many unique initial states get sampled, so if you run with default batch_size every episode in libero_10 starts from the same few configurations. longer horizon tasks are way more sensitive to this than libero_object which is why your scores diverge so much between the two.

try `--eval.n_episodes=50 --eval.batch_size=1` on libero_10 specifically and see if it comes up.

**Comment by jashshah999** [comment]
For reproducing and comparing checkpoint results, [lerobot-bench](https://github.com/jashshah999/lerobot-bench) can help systematically verify whether different checkpoints perform as expected:

```bash
pip install lerobot-bench

# Compare official checkpoint against your finetuned one
lerobot-bench compare HuggingFaceVLA/smolvla_libero ./my_smolvla_finetuned \
  --env libero --task libero_10 --n-episodes 50 --seeds 1000,2000,3000

# Or just analyze existing eval results
lerobot-bench analyze eval_official/eval_info.json eval_mine/eval_info.json
```

Reports confidence intervals and statistical significance — helps distinguish "model is broken" from "variance across seeds."

https://github.com/jashshah999/lerobot-bench

**Comment by Rick0525** [comment]
Heads-up on one axis that can silently shift these numbers: the mujoco version. From mujoco 3.4.0 the stored libero_spatial task-5 init states no longer settle onto the ramekin, which alone moves SmolVLA's spatial score by ~3.6 pp suite-level (80% → 28% on that task). Details and a no-GPU probe: #4390.

Note this does **not** explain the numbers reported above: 3.3.2 is on the healthy side of the boundary, so at least that axis can be ruled out for the original report. But anyone landing here after a fresh install (which resolves to mujoco 3.8.1 today) should check it first.
